### PR TITLE
CP-4493 app banner ist not triggered

### PR DESCRIPTION
**Pull request by jprajapati1996**

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/Constants.java [hunk 1]
```diff
@@ -2,4 +2,9 @@
 
 public interface Constants {
     String LOG_TAG = "CleverPush";
+
+    // USE FOR CHECK VERSION
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/Constants.java:6]: do we still use this?

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
 import android.os.Handler;
 import android.os.HandlerThread;
-import android.util.Log;
 import android.view.View;
 
 import com.cleverpush.ActivityLifecycleListener;
 import com.cleverpush.CleverPush;
 import com.cleverpush.CleverPushHttpClient;
 import com.cleverpush.CleverPushPreferences;
+import com.cleverpush.Constants;
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:17]: not needed

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 3]
```diff
@@ -355,61 +356,65 @@ private boolean appVersionFilter(boolean allowed, Banner banner) {
...
             e.printStackTrace();
             Logger.e(TAG, "Error checking app version filter", e);
         }
+
         return allowed;
     }
 
-    private boolean checkRelationFilter(boolean allowed, CheckFilterRelation relation, String versionName, String compareValue) {
+    private boolean checkRelationFilter(boolean allowed, CheckFilterRelation relation, String appVersion, String compareValue, String fromVersion, String toVersion) {
         VersionComparator vc = new VersionComparator();
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:369]: not sure if we want to shorten this @gbhrdt

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 4]
```diff
@@ -355,61 +356,65 @@ private boolean appVersionFilter(boolean allowed, Banner banner) {
...
-                if (mainValue != 0) {
+                if (result != VersionComparison.EQUALS) {
                     allowed = false;
                 }
             }
 
-            if (allowed && relation.equals(CheckFilterRelation.NotEqual)) {
-                if (mainValue == 0) {
+            if (allowed && relation.equals(CheckFilterRelation.Between)) {
+                if (resultFrom != VersionComparison.EQUALS && resultFrom != VersionComparison.GREATER_THAN && resultTo != VersionComparison.EQUALS && resultTo != VersionComparison.GREATER_THAN) {
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:387]: multiple lines please
Also if its equal, is it inbetween? not sure how we define this @gbhrdt

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java [hunk 5]
```diff
@@ -25,13 +25,13 @@ public int compare(Object o1, Object o2) {
...
                         // Version one is longer than number two, and non-zero
-                        return 1;
+                        return VersionComparison.GREATER_THAN;
                     }
                 }
                 while (tokenizer1.MoveNext());
 
                 // Version one is longer than version two, but zero
-                return 0;
+                return VersionComparison.EQUALS;
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java:34]: not sure if Equals is correct here, can you give some examples here?

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java [hunk 6]
```diff
@@ -76,15 +76,15 @@ public int compare(Object o1, Object o2) {
...
                     // Version one is longer than version two, and non-zero
-                    return -1;
+                    return VersionComparison.LESS_THAN;
                 }
             }
             while (tokenizer2.MoveNext());
 
             // Version two is longer than version one, but zero
-            return 0;
+            return VersionComparison.EQUALS;
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java:85]: not sure if Equals is correct here, can you give some examples here?

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java [hunk 7]
```diff
@@ -76,15 +76,15 @@ public int compare(Object o1, Object o2) {
...
         }
-        return 0;
+        return VersionComparison.EQUALS;
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java:87]: not sure if Equals is correct here, can you give some examples here?

**Review by cpKevinrodriguesnobre — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/Constants.java (hunk 1 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/Constants.java:6]: Not using just removed

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 2 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:17]: removed

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 4 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:387]: here we are checking all conditions if the current sdkVersion is in between the required version or not.

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java (hunk 5 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java:34]: here if we get the 0 from this method it means that both the versions are Equal.

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java (hunk 6 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java:85]: here if we get the 0 from this method it means that both the versions are Equal.

File: cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java (hunk 7 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/util/VersionComparator.java:87]: here if we get the 0 from this method it means that both the versions are Equal.

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 3 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:369]: here only added fromVersion and toVersion from my end

**Pull request merged**